File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.scala
object Solution {
    def lengthOfLongestSubstring(s: String): Int = {
        val last = scala.collection.mutable.Map[Char, Int]()
        var res, start = 0
        for ((c, i) <- s.zipWithIndex) {
            if (last.contains(c) && last(c) >= start) 
                start = last(c) + 1
            res = res.max(i - start + 1)
            last(c) = i
        }
        res
    }
}

File: sol/solution/0001-0100/0003. Longest Substring Without Repeating Characters/Solution.dart
class Solution {
  int lengthOfLongestSubstring(String s) {
    final last = <String, int>{};
    int res = 0, start = 0;
    for (int i = 0; i < s.length; i++) {
      final c = s[i];
      if (last.containsKey(c) && last[c]! >= start) {
        start = last[c]! + 1;
      }
      res = (i - start + 1 > res) ? i - start + 1 : res;
      last[c] = i;
    }
    return res;
  }
}